.github/workflows/ci.yml

```diff
@@ -14,6 +14,25 @@ jobs:
       run: sudo apt-get install autoconf automake pkg-config libevent-dev libpcre3-dev libssl-dev
     - name: Build
       run: autoreconf -ivf && ./configure && make
+    - name: Install test dependencies
+      run: |
+        git clone git://github.com/antirez/redis.git --branch 5.0.7
+        cd redis && make malloc=libc && make && cd ..
+        ./redis/src/redis-server --version
+        sudo apt-get install python3-pip 
+        sudo apt-get install python3-setuptools
```

Comment:
GitHub Actions has a more direct support for Python envs.
You should use something like:
```
    - name: Setup Python
      uses: actions/setup-python@v1
      with:
        python-version: '3.x'
        architecture: 'x64'
```

Then you can just run `pip install`.
And if you want to take it another step, you should also consider caching some steps to make this run a bit faster (probably not a big deal though).

Reply:
updated to use direct python support. also caching dependencies as you've stated :)